### ISSUE TITLE
What to do about "usable" entities?

### ISSUE THREAD

**Issue by bmorel** [Unvanquished/Unvanquished#2047]
We did various changes in 0.53 regarding how "usable" entities work.
At this time, the only entity which was kept in mind was the armoury.

What we discovered recently is that damage-triggered entities in unvanquished behave in fact as "usable" entities, which is obviously a bug, but makes maps using it usable, at least. The aforementioned changes change this fact, making them unusable, not because the bug is fixed, but because of how the algorithm was changed.

PRs #1837 and #1952 both aim at improving on the situation, but I'm very tempted to just close them until we decide what to do, once and for all.

Things to discuss are:

* how should the UI behave? This includes the usage of commands to buy or sell stuff.
* is it acceptable to keep damage-triggered entities "usable" until we get a real implementation?
* should we implement properly such things as "usable" doors and buttons?
* last but not least, should all the problems be fixed in a single PR or not?

**Comment by bmorel** [comment]
My opinion is:

* things should only be usable if you're actually able to see them, and if not, commands to buy/sell should fail;
* it is acceptable to keep a bug that allows maps to be playable;
* we should implement an explicit, non-buggy, way to have doors and buttons usable
* we should not do everything in a single PR

**Comment by bmorel** [comment]
pinging some people around, that I think are quite interested by the question for various reasons: @illwieckz , @slipher , @cu-kai  and @DolceTriade 

This is to replace the discussions I think will be fruitless or a waste of time if done on individual issues or PRs.

**Comment by cu-kai** [comment]
As far as using things like armouries goes, we should maintain the legacy behaviour (and actually, I like the current behaviour too) where they can be used through walls. Being (now) able to open an armoury though a wall with `+activate+` is very useful and indeed lowers the skill gap caused by players having binds.

I think there should _also_ be a HUD tip for those with the tutorial on for "using" an entity - and this should ideally be in a bright colour (such as green, `^2`) similar to the new medkit tip when you are low on health.

I would like to see usable doors/entities in the game as well, though we need to implement this in a way which does not break existing behaviour (e.g., actually fix the real damage buttons so they work as they did in Tremulous, and finally fix #843) - this means only new maps made or updated from that time onwards will be able to make use of this feature.

**Comment by DolceTriade** [comment]
My opinion is: It's really not for us to decide whether something is usable or not (ie, by mandating that it be visible). I think that mappers/modders should be able to make any collidable entity usable.

We should provide some indication that there is a usable entity in the vicinity, either via the hand icon, or a beacon.

I think the hand icon should also have some text that says `Press <look up bind key for use>`

**Comment by bmorel** [comment]
> My opinion is: It's really not for us to decide whether something is usable or not (ie, by mandating that it be visible). I think that mappers/modders should be able to make any collidable entity usable.

So we should remove the hand icon when arriving near armory?

Another important question about those (armory being one of those) is: should them be usable when they are in the back of the player (with binds, like what is currently doable with armory) or when not aiming at said entity (as currently with armory)?

**Comment by DolceTriade** [comment]
Fair point. Perhaps we can make a flag or something that allows some entities to be marked as usable.

**Comment by bmorel** [comment]
Keep in mind that the question is not only about armory, but about any usable entity. I insist on this because I'd rather have the code as generic as possible. We already suffer much about code duplication and exceptions for not much.
To be able to buy even without LoS also gives a huge advantage to "those who know" (those who know how to create binds), since we currently do not have default bindings nor any interface to build such binds.
Even the GUI does not provides a sensible way to buy the best armor possible...

**Comment by DolceTriade** [comment]
I'm ok with a set of players having an advantage because they know about the game more.
